ui-v2/app/instance-initializers/nspace.js

```diff
@@ -1,11 +1,107 @@
+        return this._super(...withNspace(this.routeName, requestedRouteName, ...rest));
+      };
+      return prev;
+    }, {})
+  );
+
+  // patch Route routeName-like methods for data to support nspace relative routes
+  Route.reopen(
+    ['modelFor', 'paramsFor'].reduce(function(prev, item) {
+      prev[item] = function(requestedRouteName, ...rest) {
+        const isNspaced = this.routeName.startsWith('nspace');
+        if (!isNspaced && requestedRouteName === 'nspace') {
+          return {
+            nspace: '~',
+          };
+        }
+        return this._super(...withNspace(this.routeName, requestedRouteName, ...rest));
+      };
+      return prev;
+    }, {})
+  );
+
+  // extend router service with a nspace aware router to support nspace relative routes
+  const nspacedRouter = container.resolveRegistration('service:router').extend({
+    transitionTo: function(requestedRouteName, ...rest) {
+      return this._super(...withNspace(this.currentRoute.name, requestedRouteName, ...rest));
+    },
+    replaceWith: function(requestedRouteName, ...rest) {
+      return this._super(...withNspace(this.currentRoute.name, requestedRouteName, ...rest));
+    },
+    urlFor: function(requestedRouteName, ...rest) {
+      return this._super(...withNspace(this.currentRoute.name, requestedRouteName, ...rest));
+    },
+  });
+  container.register('service:router', nspacedRouter);
   if (env('CONSUL_NSPACES_ENABLED')) {
+    // enable the nspace repo
     ['dc', 'settings', 'dc.intentions.edit', 'dc.intentions.create'].forEach(function(item) {
       container.inject(`route:${item}`, 'nspacesRepo', 'service:repository/nspace/enabled');
       container.inject(`route:nspace.${item}`, 'nspacesRepo', 'service:repository/nspace/enabled');
     });
     container.inject('route:application', 'nspacesRepo', 'service:repository/nspace/enabled');
+
+    const dotRe = /\./g;
+    // register automatic 'index' routes and controllers that start with 'dc'
+    Object.keys(flat(routes))
+      .filter(function(item) {
+        return item.startsWith('dc');
+      })
+      .map(function(item) {
+        return item.replace('._options.path', '').replace(dotRe, '/');
+      })
+      .forEach(function(item) {
+        let route = container.resolveRegistration(`route:${item}`);
+        let indexed;
+        // if the route doesn't exist it probably has an index route instead
+        if (!route) {
+          item = `${item}/index`;
+          route = container.resolveRegistration(`route:${item}`);
+        } else {
+          // if the route does exist
+          // then check to see if it also has an index route
+          indexed = `${item}/index`;
+          const index = container.resolveRegistration(`route:${indexed}`);
+          if (typeof index !== 'undefined') {
+            register(container, index, indexed);
+          }
+        }
+        register(container, route, item);
+      });
+
+    // tell the view we have nspaces enabled
```

Comment:
A lot of this is taken from the `initializer/nspace` file which we've now deleted after we realized we could add all this nspacing work into a single place.